src/translator.cc

```diff
@@ -352,4 +352,24 @@ namespace ctranslate2 {
     return _model->device();
   }
 
+  int Translator::device_index() const {
+    return _model->device_index();
+  }
+
+  ComputeType Translator::compute_type() const {
+    return _model->compute_type();
+  }
+
+  void Translator::set_model(const std::string& model_dir)
+  {
+    std::shared_ptr<const models::Model> model;
+    _model = model->load(model_dir, this->device(), this->device_index(), this->compute_type());
```

Comment:
You could call the other overload here, e.g.:

```cpp
set_model(model->load(model_dir, this->device(), this->device_index(), this->compute_type()));
```

Reply:
Right, done!